api/gateway/apimiddleware/process_request.go

```diff
@@ -77,7 +77,8 @@ func (m *ApiProxyMiddleware) PrepareRequestForProxying(endpoint Endpoint, req *h
 // ProxyRequest proxies the request to grpc-gateway.
 func ProxyRequest(req *http.Request) (*http.Response, ErrorJson) {
 	// We do not use http.DefaultClient because it does not have any timeout.
-	netClient := &http.Client{Timeout: time.Minute * 2}
+	netClient := &http.Client{Timeout: cmd.ApiTimeout.Value}
```

Comment:
Won't this always be 2 minutes? We usually get flag values using the cli context methods.

Reply:
Example of fetching a Duration value from a flag: https://github.com/prysmaticlabs/prysm/blob/d9b98e99134e82bc5f7c01d9d7578d43c2419cdc/validator/node/node.go#L436

Reply:
+1

Reply:
Thanks! Happy to be wrong here. The way we are currently using with the c.cliCtx.Duration is too cumbersome anyway. This is a welcome change.